### PR TITLE
Fixed #52 Adding duplicates to Apple Events

### LINKED ISSUE #52: Duplicate apps should not be added to Apple Events
Duplicate apps can be added to the Apple Events view for each app in the Applications view.

#39 was partially fixed by #43 but the Apple Events view is still able to receive duplicates. The same definition works:
The app should not allow duplicate applications. Duplicates should be defined as an app that matches the name/identifier and code requirements.

### PR DESCRIPTION
**Pull request by ty-wilson**
Added a check before an Apple Event is added to see if that rule already exists by checking for the destination among the currently selected Executable's Apple Event rules.

Fixed #52

### CODE REVIEW

File: Source/View Controllers/TCCProfileViewController.swift [hunk 1]
```diff
@@ -351,14 +351,26 @@ class TCCProfileViewController: NSViewController {
...
     }
     
     func shouldExecutableBeAdded(_ executable: Executable) -> Bool {
         return self.model.selectedExecutables.firstIndex(of: executable) == nil
     }
     
+    func shouldAppleEventRuleBeAdded(_ rule: AppleEventRule) -> Bool {
+        var found = false
+        let selectedExe = self.model.selectedExecutables[self.executablesAC.selectionIndex]
+        for existingRule in selectedExe.appleEvents {
```
Comment (macblazer) [inline Source/View Controllers/TCCProfileViewController.swift:367]: This is better done with the `.first(where:)` function like so:

```
        let foundRule = selectedExe.appleEvents.first {
            $0.destination == rule.destination
        }

        return foundRule == nil
```

File: Source/View Controllers/TCCProfileViewController.swift [hunk 2]
```diff
@@ -351,14 +351,26 @@ class TCCProfileViewController: NSViewController {
     func insetIntoAppleEvents(_ executable: Executable) {
         guard let source = self.executablesAC.selectedObjects.first as? Executable else { return }
         let rule = AppleEventRule(source: source, destination: executable, value: false)
-        guard self.appleEventsAC.canInsert else { return }
-        self.appleEventsAC.insert(rule, atArrangedObjectIndex: 0)
+        if (self.shouldAppleEventRuleBeAdded(rule)) {
```
Comment (macblazer) [inline Source/View Controllers/TCCProfileViewController.swift:354]: This should be combined with the guard on the next line.

```
guard self.appleEventsAC.canInsert,
    self.shouldAppleEventRuleBeAdded(rule) else {
    return
}
```

File: Source/View Controllers/TCCProfileViewController.swift [hunk 3]
```diff
@@ -351,14 +351,26 @@ class TCCProfileViewController: NSViewController {
     func insetIntoAppleEvents(_ executable: Executable) {
```
Comment (macblazer) [inline Source/View Controllers/TCCProfileViewController.swift:351]: Just noticed this should be renamed to `insertIntoAppleEvents` (missing the `r` currently).

Here's a chance to try out Xcode's Refactor > Rename functionality.  Select this function name, right click, and choose Refactor > Rename... to make sure it's updated everywhere it is used.

**Comment by ty-wilson** [comment]
Awesome, thanks for cleaning that up for me!

**Pull request merged**